Add Mic Editor

joshuadahlunr opened the pull request:
Added a custom GUI to the Mic class which allows selecting the input device to use from within the Unity Editor.

Commit:
Add Mic Editor

Added a custom GUI to the Mic class which allows selecting the input device to use from within the Unity Editor.

adrenak commented on Assets/UniMic/Runtime/Editor/MicEditor.cs:
```diff
@@ -0,0 +1,58 @@
+			EditorGUILayout.EndHorizontal();
+
+			// Display a button to reset to the system's default mic
+			if (EditorGUILayout.LinkButton("Switch to System's Default Device"))
```
Change EditorGUILayout.LinkButton to GUILayout.Button as it is not available on pre Unity 2021

adrenak commented on Assets/UniMic/Runtime/Editor/MicEditor.cs:
```diff
@@ -0,0 +1,58 @@
+
+			// Display a button to reset to the system's default mic
+			if (EditorGUILayout.LinkButton("Switch to System's Default Device"))
+				mic.ChangeDevice(0);
```
Change from ChangeDevice to SetDeviceIndex

A little change to make the methods purpose clearer.

adrenak committed:
skip ci

- Change EditorGUILayout.LinkButton to GUILayout.Button
- Change ChangeDevice to SetDeviceIndex

adrenak committed:
Merge branch 'master' into MicEditor
